sdk/cosmosdb/cosmos/test/internal/unit/sasToken.spec.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import assert from "assert";
+import { CosmosClient } from "../../../src";
+import { endpoint, masterKey } from "../../public/common/_testConfig";
+import { SasTokenPermissionKind } from "../../../src/common";
+
+describe.only("SAS Token Authorization", function() {
+  const sasTokenClient = new CosmosClient({
+    endpoint,
+    key: masterKey,
+    sasTokenProperties: {
+      user: "user1",
+      userTag: "",
+      databaseName: "db1",
+      containerName: "coll1",
+      resourcePath: "/dbs/db1/colls/coll1/",
+      partitionKeyValueRanges: [],
+      startTime: new Date(),
+      expiryTime: new Date(),
+      keyType: 0,
+      controlPlaneReaderScope: SasTokenPermissionKind.ContainerReadAny,
+      controlPlaneWriterScope: 0,
+      dataPlaneReaderScope: SasTokenPermissionKind.ContainerFullAccess,
+      dataPlaneWriterScope: 0
+    }
+  });
+
+  it("should connect with sas token properties set", async function() {
+    assert(
+      undefined !== (await sasTokenClient.databases.readAll().fetchAll()),
```

Comment:
couldn't this error yet still not be undefined?

Reply:
Experimental feature.

Reply:
sounds good but definitely want this to change going forward